### PR TITLE
Merge global storage options on create

### LINKED ISSUE #34074: Cannot change default storage size
On Windows 10 I tried to change the default storage size of containers by configuring daemon.json and adding an option "storage-opts": ["size=50G"]. This does not have any effect. 

When I execute `docker run microsoft/nanoserver powershell -c get-psdrive` I would expect to have a C drive with 50G instead it has default 20G. When I specify storage-opt for individual docker run it works. However this is not a solution for me because I need to have more disk space during a build process.

docker version:
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:30:30 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.24)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 22:19:00 2017
 OS/Arch:      windows/amd64
 Experimental: false

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard <jhoward@microsoft.com>

Fixes https://github.com/moby/moby/issues/34074 and https://github.com/docker/for-win/issues/999.

Ensures the daemons global storage options are merged into the user requested options. Required to ensure that the sandbox size is passed through to the driver (this applies to both WCOW and LCOW). See screenshot in https://github.com/docker/for-win/issues/999#issuecomment-322263860 for demonstration of this fixing the issue.

ping @thaJeztah @johnstep @darrenstahlmsft 

cc @vikalyan & @rustyCZ

### CODE REVIEW

File: daemon/create.go [hunk 1]
```diff
@@ -137,6 +137,23 @@ func (daemon *Daemon) create(params types.ContainerCreateConfig, managed bool) (
 
 	container.HostConfig.StorageOpt = params.HostConfig.StorageOpt
 
+	// Fixes: https://github.com/moby/moby/issues/34074 and
+	// https://github.com/docker/for-win/issues/999.
+	// Merge the daemons storage options if they aren't already present. We only
```
Comment (johnstep) [inline daemon/create.go]: Nit: daemon's

**Review by johnstep — APPROVED** [review]
LGTM

File: daemon/create.go (hunk 1 above)
Comment (jhowardmsft) [inline daemon/create.go]: Punctuation police ;) Yeah, fixed.

File: daemon/create.go [hunk 2]
```diff
@@ -137,6 +137,23 @@ func (daemon *Daemon) create(params types.ContainerCreateConfig, managed bool) (
...
+	// Merge the daemon's storage options if they aren't already present. We only
+	// do this on Windows as there's no effective sandbox size limit other than
+	// physical on Linux.
+	if runtime.GOOS == "windows" {
+		if container.HostConfig.StorageOpt == nil {
+			container.HostConfig.StorageOpt = make(map[string]string)
+		}
+		for _, v := range daemon.configStore.GraphOptions {
+			opt := strings.SplitN(v, "=", 2)
+			if _, ok := container.HostConfig.StorageOpt[opt[0]]; !ok {
```
Comment (thaJeztah) [inline daemon/create.go]: If Should we explicitly limit this to only the `size` option? `--storage-opt` on the daemon is also used to initialise the storage-driver, so may include options that cannot be used per-container, or is that not a concern?

**Comment by thaJeztah** [comment]
This needs an update to the daemon docs as well (in the docker/cli repository); https://github.com/docker/cli/blob/master/docs/reference/commandline/dockerd.md

File: daemon/create.go (hunk 2 above)
Comment (jhowardmsft) [inline daemon/create.go]: We don't need to be concerned with this on either of the Windows graphdrivers.

Nice spelling of initialise though ;)

**Comment by jhowardmsft** [comment]
> This needs an update to the daemon docs as well (in the docker/cli repository); https://github.com/docker/cli/blob/master/docs/reference/commandline/dockerd.md

Yeah, that can be a follow up PR 😇

**Comment by vikalyan** [comment]
@jhowardmsft Super! How do I get this build on Win2k16 server?

File: daemon/create.go (hunk 2 above)
Comment (thaJeztah) [inline daemon/create.go]: > Nice spelling of initialise though ;)

🇬🇧:+1:

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by jhowardmsft** [comment]
When merged it will be on master.dockerproject.org shortly after

**Comment by jhowardmsft** [comment]
Jenkins or Leeroy are playing up but started RS1 manually @ https://jenkins.dockerproject.org/job/Docker-PRs-WoW-RS1/16302/console

**Comment by vikalyan** [comment]
@jhowardmsft  Sounds good. Looking forward to trying this out tomorrow! Thanks for the quick turnaround!

**Comment by jhowardmsft** [comment]
Success. Just not showing here for some reason. 

00:56:10.411 Set build name.
00:56:10.412 New build name is '#34508'
00:56:10.416 Variable with name 'BUILD_DISPLAY_NAME' already exists, current value: '#34508', new value: '#34508'
00:56:11.530 [PostBuildScript] - Execution post build scripts.
00:56:11.531 Notifying endpoint with url 'https://leeroy.dockerproject.org/notification/jenkins'
00:56:12.141 Finished: SUCCESS

**Comment by thaJeztah** [comment]
Works for me 👍

**Comment by thaJeztah** [comment]
@jhowardmsft can you do a follow-up for the documentation?

**Comment by jhowardmsft** [comment]
@thaJeztah https://github.com/docker/cli/pull/442

**Comment by PatrickLang** [comment]
We need to get this into a Docker-EE release. Today, the default scratch space has a hard limit of 20GB, and if anyone needs to build containers that will exceed 20GB, they need to override it. 

Problems we've heard:

**Build Workers**
If you try to install big apps like the Visual Studio tools along with the full Windows SDKs, then you will run out of space during the `docker build` process. This could be a blocker for customers wanting to build containers with the right tools for Jenkins, appveyor, VSTS, and so on.

**Comment by carlfischer1** [comment]
@PatrickLang thanks - I've opened an internal tracking issue

**Pull request merged**